added documentation on leaderworkerset features (#67)
kubernetes-sigs/lws: docs/examples/sample/README.md

> # LeaderWorkerSet Features

Reviewer: maybe: "How to use LeaderWorkerSet Features"

> LeaderWorkerSet supports the following features.

Reviewer: "Below are some examples to use different features of LeaderWorkerSet"

> 

Reviewer: Add one more example before these.

"Deploy a LeaderWorkerSet with 3 groups and 4 workers in each group"

And add the example yaml as well.

## Multi template for leader/worker pods

> Different templates for leaders and workers can be specified as seen [here](lws-multi-template.yaml),

Reviewer: ```suggestion
LWS support using different templates for leader and worker pods. You could find the example [here](lws-multi-template.yaml), leader pod's spec is specified in leaderTemplate, and worker pods' spec is specified in workerTemplate.
```

Leader template is an optional field. If not specified, it will default to the worker template.

## Restart Policy

> There exists an API field called RestartPolicy to define the failure handling schematics for the pod group.

Reviewer: ```suggestion
You could specify the RestartPolicy to define the failure handling schematics for the pod group.
```

> On default, only failed pods will be restarted. When it is set to RecreateGroupOnRestart, it will recreate 

Reviewer: ```suggestion
By default, only failed pods will be automatically restarted. When the RestartPolicy is set to RecreateGroupOnRestart, it will recreate 
```

> the groups on container/pods restart. All the worker pods will be stopped before the new leader pod is started.

Reviewer: ```suggestion
the groups on container/pods restarts. All the worker pods will be recreated after the new leader pod is started.
```

> This field as set as shown [here](lws-restart-policy.yaml)

Reviewer: ```suggestion
You could find an example [here](lws-restart-policy.yaml)
```

## Horizontal Pod Autoscaler (HPA)

> A horizontal pod autoscaler can be deployed alongside LeaderWorkerSet. An example yaml can be found [here](horizontal-pod-autoscaler.yaml)

Reviewer: ```suggestion
LWS expose a scale endpoint for HPA to trigger scaling. An example HPA yaml for LWS can be found [here](horizontal-pod-autoscaler.yaml)
```

## Exclusive Placement

> LeaderWorkerSet can force exclusive 1:1 placement so that each Leader Worker group is scheduled on the same node pool 

Reviewer: ```suggestion
LeaderWorkerSet support exclusive placement through node affinity/anti-affinity where pods in the same group will be scheduled on the same topology domain (ex, GKE node pool), but on different nodes. (In the context of GKE node pool, this ensures 1:1 pod-to-node mapping and 1:1 podgroup-to-nodepool mapping.
```

> by adding the exclusive topology annotation as shown [here](lws-exclusive-placement.yaml)

Reviewer: ```suggestion
This feature can be enabled by adding the exclusive topology annotation `leaderworkerset.sigs.k8s.io/exclusive-topology:` as shown [here](lws-exclusive-placement.yaml)
```
